lib/retrieve_title.rb

```diff
@@ -60,6 +60,10 @@ def self.fetch_title(url)
     encoding = nil
 
     fd.get do |_response, chunk, uri|
+      if (uri.present? && InlineOneboxer.domain_is_blocked?(uri.hostname))
+        throw :done
```

Comment:
tricky... but I am not sure if this is technically correct, we only want to apply the ban hammer on the last redirect in the chain, so if you work through a blocked onebox domain on your way to a non blocked one, it is fine.

Reply:
I wrote a test and it fulfils what you said. I'll add it in as well.

The `FinalDestination` class only returns the ... final destination, and it's what we're using here.